antenna/ext/pubsub/crashpublish.py

```diff
@@ -0,0 +1,75 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import logging
+
+from everett.component import ConfigOptions
+from google.cloud import pubsub_v1
+
+from antenna.ext.crashpublish_base import CrashPublishBase
+from antenna.heartbeat import register_for_verification
+
+
+logger = logging.getLogger(__name__)
+
+
+class PubSubCrashPublish(CrashPublishBase):
+    """Publisher to Pub/Sub.
+
+    This does **not** create a topic if one does not exist. Instead, it'll
+    throw errors when it tries to publish to a topic that does not exist.
+    Whoever sets up infrastructure is in charge of creating the topic.
+
+    This will retry publishing several times, then give up. If it gives up, the
+    crashpublisher will put the crash id back in the queue to retry again
+    later.
+
+    This creates a ``PublisherClient`` which will connect to a local Pub/Sub
+    emulator if ``PUBSUB_EMULATOR_HOST=host:port`` is defined in the
+    environment. Otherwise, it'll connect to Google Cloud Pub/Sub.
+
+    """
+
+    required_config = ConfigOptions()
+    required_config.add_option(
+        'project_id',
+        doc='Google Cloud project id.'
+    )
+    required_config.add_option(
+        'topic_name',
+        doc='The Pub/Sub topic name to publish to.'
+    )
+
+    def __init__(self, config):
+        super().__init__(config)
+
+        self.project_id = self.config('project_id')
+        self.topic_name = self.config('topic_name')
+
+        # Batches crash_ids for up 1024 bytes or a second
+        self.batch_settings = pubsub_v1.types.BatchSettings(
+            max_bytes=1024,
+            max_latency=1,
+        )
```

Comment:
Since crash ids are small, I think we won't hit `max_bytes`. I think we'll hit the 1-second latency for batching. That's probably fine. If we find we want to tune that, we can make it configurable.